### PR TITLE
feature/AT-6532 > Add authentication to ATAT Web UI via Cognito

### PR DESCRIPTION
**Pull request by tonysumner-ccpo**
This ticket adds the ability to authenticate users from ATAT Web UI against our Azure AAD by way of the Cognito configuration created in AT-6531. Although it does not require authorization to be implemented, it should provide the ability to restrict access to the UI to authenticated users.

### CODE REVIEW

**Comment by tonysumner-ccpo** [comment]
Have implemented the button click. However, the results of  the button click are below.   It looks as if I missing a `user pool or identity pool in config`

image

**Comment by jasonburkert-ccpo** [comment]
This should be unblocked now that #123 has been merged.

**Comment by kylelaker-ccpo** [comment]
I rebased this branch, so when you start on this ticket, do not pull. You'll have to do something a little more special here. If you do a regular pull and merge commit there will be a bunch of duplicate commits. Finish work on your existing branch and commit/push all changes (because an upcoming command will cause you to lose them)

1. `git stash` (just in case there's anything left)
1. `git fetch --all`
1. ` git switch feature/AT-6532`
1. `git reset --hard origin/feature/AT-6532`

File: src/store/index.ts [hunk 1]
```diff
@@ -155,22 +155,40 @@ export default new Vuex.Store({
...
+      state.user.dod_id = user?.["custom:dod_id"] ?? "1234567890";
+      state.user.citizenship = user?.["custom:citizenship"] ?? "United States";
+      state.user.designation = user?.["custom:designation"] ?? "Civilian";
+      // This field may not be available from our identity provider
+      state.user.phone_number = user?.phone ?? "(555) 555-5555";
+      // There is not currently a known way to get this information from
+      // the identity provider.
+      state.user.service_branch = "U.S. Army";
+      state.user.title = "Ms.";
+    },
```
Comment (kylelaker-ccpo) [inline src/store/index.ts:191]: I think my changes in this file are the most bad of my changes and the ones where I really didn't know if this was the right way to do it.

**Comment by kylelaker-ccpo** [comment]
@tonysumner-ccpo I rebased this again to address the merge confict

**Review by robertmccardell-ccpo — APPROVED** [review]

**Pull request merged**